lib/streamlit/connections/snowflake_connection.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: We won't always be able to import from snowflake.{connector, snowpark}.* so need
+# the `type: ignore` comment below, but that comment will explode if `warn-unused-ignores`
+# is turned on when the package is available. Unfortunately, mypy doesn't provide a good
+# way to configure this at a per-line level :(
+# mypy: no-warn-unused-ignores
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, Sequence, cast
+
+import pandas as pd
+
+from streamlit.connections import BaseConnection
+from streamlit.connections.util import load_from_snowsql_config_file, running_in_sis
```

Comment:
Note to snowflake drivers reviewers: Worth taking a look at the implementations of these methods too (outside this PR) and seeing if you agree - in particular `running_in_sis()` is important!

Reply:
For `running_in_sis` you should use the following snowpark utility method instead https://github.com/snowflakedb/snowpark-python/blob/697294d8ee89396deab4309dea384b1eadb75d6b/src/snowflake/snowpark/_internal/utils.py#L500-L501
We're in the process of renaming the SP connection to match the public one.

Reply:
👍 will do